### DISCUSSION TITLE
How to get sub-elements of elements? how to further select inner HtmlElement of HtmlElement?

### DISCUSSION THREAD

**Discussion by FurkanGozukara** [cefsharp/CefSharp.Dom#46] [Q&A]
Hello. I am getting multiple elements like this :

`   var elements = await devToolsContext.QuerySelectorAllAsync<HtmlElement>("div.vehicle-card");`

It returns a result like below

![image](https://user-images.githubusercontent.com/19240467/206589419-ed0f8bb6-56c1-4a06-88a3-5e70a6051e81.png)

Now I want to iterate through elements inside elements  like this and get specific information

For example

```
            foreach (var element in elements)
            {
              var title= await  element.GetChildNodesAsync<HtmlElement>("h2.title").GetInnerTextAsync();
            }
```

but this code is not working. So I want to further select inner element of selected HtmlElement

**Comment by FurkanGozukara** [comment]
@amaitland  thank you for help. I didn't ask this on stackoverflow. 

I made it work like this but i am not sure if it is the proper way

     foreach (var element in elements)
            {
                var selectedTitle = await element.QuerySelectorAsync("h2.title");
                var title = await selectedTitle.GetTextContentAsync();
            }

**by the way the selectedTitle  doesnt have GetInnerTextAsync as shown in example**

**Reply by amaitland** [reply]
> by the way the selectedTitle doesnt have GetInnerTextAsync as shown in example

You need to specify the generic type argument. 

The examples in the readme are generated from actual code, there's compile time checking.
